### PR TITLE
#1416 list devices under site

### PR DESCRIPTION
**Pull request by mspratt-biot**
image

### CODE REVIEW

File: udmif/api/src/__tests__/dao/DAO.spec.ts [hunk 1]
```diff
@@ -37,60 +38,85 @@ describe('DAO', () => {
     await connection.close();
   });
 
-  test('getAll', async () => {
-    const insertedDeviceDocument: Device = { id: 'd-id-1', name: 'device-1', site: 'LOC' };
+  describe('getAll', () => {
```
Comment (mspratt-biot) [inline udmif/api/src/__tests__/dao/DAO.spec.ts:41]: Just changed all these to describe blocks.

File: udmif/api/src/__tests__/dao/DAO.spec.ts [hunk 2]
```diff
@@ -37,60 +38,85 @@ describe('DAO', () => {
...
+      deviceCollection.insertMany([insertedDeviceDocument1, insertedDeviceDocument2]);
 
-    const deviceNames: string[] = await deviceDao.getDistinct('name', { limit: 10 });
+      const deviceNames: string[] = await deviceDao.getDistinct('name', { limit: 10 });
 
-    expect(deviceNames).toEqual(['device-1', 'device-2']);
+      expect(deviceNames).toEqual(['device-1', 'device-2']);
+    });
+
+    test('return unique values of a field in a document with a filter', async () => {
```
Comment (mspratt-biot) [inline udmif/api/src/__tests__/dao/DAO.spec.ts:107]: Added this test due to new feature being able to get distinct values based on filtered results.

File: udmif/api/src/mongo/MongoFilterBuilder.ts [hunk 3]
```diff
@@ -30,5 +30,5 @@ function convertMapOfFiltersToFilter(mapOfFilters: Map<string, Filter[]>): any {
     })
     .filter((expression) => expression);
 
-  return { $or: filterExpressions };
+  return { $and: filterExpressions };
```
Comment (mspratt-biot) [inline udmif/api/src/mongo/MongoFilterBuilder.ts:33]: The or behaviour just seemed so weird, I decided to change it to and.

Comment (slevertbiot) [reply udmif/api/src/mongo/MongoFilterBuilder.ts:33]: and ...  😉

**Review by slevertbiot — APPROVED** [review]

**Pull request merged**